HW-5 учимся реквестить!

AndervolV opened the pull request.

AndervolV committed:
HW-5 учимся реквестить!

dmitry-bizin commented on src/Main.java:
```diff
@@ -13,5 +13,96 @@ public static void main(String[] args) {
+                System.out.println("Установите облегченную версию приложения для Android по ссылке");
+            }
+        } else if (clientDeviceYear > 2015) {
+            if (clientOS == 0) {
```
хотел не плодить переменные, но в итоге всё равно наплодил) здесь у тебя другая переменная используется

dmitry-bizin replied:
вложенность есть в коде, согласно заданию её быть не должно

dmitry-bizin commented on src/Main.java:
```diff
@@ -13,5 +13,96 @@ public static void main(String[] args) {
         }
+        System.out.println("Задача N2");
+//Задача 2 Чтобы не плодить переменные переменная clientOS взята из первой задачи
+        int clientOs = 1;
```
```suggestion
        clientOS = 1;
```
если не хотел плодить переменные, то нужно было сделать вот так

dmitry-bizin commented on src/Main.java:
```diff
@@ -13,5 +13,96 @@ public static void main(String[] args) {
+        int clientOs = 1;
+        int clientDeviceYear = 2015;
+        if (clientDeviceYear <= 2015) {
+            if (clientOs == 0) {
+                System.out.println("Установите облегченную версию приложения для iOS по ссылке");
+            } else {
+                System.out.println("Установите облегченную версию приложения для Android по ссылке");
+            }
```
вложенность есть в коде, согласно заданию её быть не должно

dmitry-bizin commented on src/Main.java:
```diff
@@ -13,5 +13,96 @@ public static void main(String[] args) {
+        }
+        System.out.println("Задача N3");
+//Задача 3
+        System.out.println("Наставник! это задача не про условные операторы, а про циклы, которые мы еще будем изучать.");
```
нет, это про условные операторы задача

dmitry-bizin commented on src/Main.java:
```diff
@@ -13,5 +13,96 @@ public static void main(String[] args) {
+//Задача 3
+        System.out.println("Наставник! это задача не про условные операторы, а про циклы, которые мы еще будем изучать.");
+        int leapYear = 2014;
+        if (leapYear >= 1584 && leapYear <= 2100) {
+            if (leapYear % 4 == 0 || leapYear % 400 == 0 && leapYear % 100 != 0) {
```
здесь будет всего 1 if и 1 else, без вложенности. Надо просто в if корректно написать условие для года: больше 1584 И делится на 4 И НЕ делится на 100 ИЛИ делится на 400

dmitry-bizin commented on src/Main.java:
```diff
@@ -13,5 +13,96 @@ public static void main(String[] args) {
+//Задача 5
+        int month = 11;
+        switch (month) {
+            case 1:
+                System.out.println("Зима");
+                break;
+            case 2:
+                System.out.println("Зима");
+                break;
+            case 3:
+                System.out.println("Весна");
+                break;
+            case 4:
+                System.out.println("Весна");
+                break;
+            case 5:
+                System.out.println("Весна");
+                break;
+            case 6:
+                System.out.println("Лето");
+                break;
+            case 7:
+                System.out.println("Лето");
+                break;
+            case 8:
+                System.out.println("Лето");
+                break;
+            case 9:
+                System.out.println("Осень");
+                break;
+            case 10:
+                System.out.println("Осень");
+                break;
+            case 11:
+                System.out.println("Осень");
+                break;
+            case 12:
+                System.out.println("Зима");
+                break;
```
кейсы лучше объединить, т.е. всего 4 кейса будет по сезонам

AndervolV committed:
HW-5 учимся реквестить!правильно ли, что при работе с поправками снова нужно отделять ветку

AndervolV committed:
HW-5 учимся реквестить!правильно ли, что при работе с поправками снова нужно отделять ветку
Мда, запутался

AndervolV committed:
HW-5 учимся реквестить!правильно ли, что при работе с поправками снова нужно отделять ветку
Мда, запутался. Сейчас скачал с Gita и делаю просто комит и пуш, но уже с исправлениями

dmitry-bizin commented on src/Main.java:
```diff
@@ -14,34 +14,30 @@ public static void main(String[] args) {
+        if (clientDeviceYear <= 2015 && clientOs == 0) {
+            System.out.println("Установите облегченную версию приложения для iOS по ссылке");
+        } else {
+            System.out.println("Установите облегченную версию приложения для Android по ссылке");
```
не, это неверно. Почему именно для Android и почему облегченная? может быть iOS и обычная. В этом задании должны быть if/else-if/else, а не просто 2 if/else

AndervolV replied:
Да, понятно. Затупил

dmitry-bizin commented on src/Main.java:
```diff
@@ -14,34 +14,30 @@ public static void main(String[] args) {
+
+            System.out.println("Устоновите версию приложения для iOS по ссылке");
+        } else {
+            System.out.println("Устоновите версию приложения для Android по ссылке");
```
аналогично

dmitry-bizin commented on src/Main.java:
```diff
@@ -14,34 +14,30 @@ public static void main(String[] args) {
-            } else {
-                System.out.println(leapYear + " год не является высокосным");
-            }
+        if (leapYear >= 1584 && leapYear % 4 == 0) {
```
условие високосного года неверное (см. задание, там написано правильное условие)

dmitry-bizin commented on src/Main.java:
```diff
@@ -61,48 +57,15 @@ public static void main(String[] args) {
         }
         System.out.println("Задача N5");
-//Задача 5
+//Задача 5 видимо так надо
```
в целом да, только этот вариант switch не рассматривался в уроке, он более новый. Можно было вот так ещё (этот был в уроке, в шпаргалке вроде, рекомендую внимательнее читать/смотреть уроки и шпаргалки, там есть вся информация):
```java
switch (month) {
    case 1:
    case 2:
    case 12:
         System.out.println("Зима");
         break;
    case 3:
    case 4:
    case 5:
         System.out.println("Весна");
         break;
    case 6:
    case 7:
    case 8:
         System.out.println("Лето");
         break;
    case 9:
    case 10:
    case 11:
         System.out.println("Осень");
         break;
    default:
        System.out.println("Нет такого месяца");
};
```

AndervolV replied:
Когда я разбирал в чем у меня проблема с "switch" Я попросил Deep объяснить как правильно пользоваться "switch" Было разобрано два варианта, Один как в курсе и один более новый вариант. Он был по короче и я решил воспользоваться им. В следующий раз буду пользоваться материалами урока

dmitry-bizin replied:
Можно использовать оба варианта) просто теперь ты знаешь, что их несколько

AndervolV committed:
HW-5 Дмитрий, Спасибо за подробные разьяснения

AndervolV commented:
Наставник во 2 задании я не понял как правильно и поставил фигурные скобки Для второго If. Жук ошибок не видит, но я думаю в каком-то случае это не корректно

dmitry-bizin commented on src/Main.java:
```diff
@@ -15,30 +15,37 @@ public static void main(String[] args) {
-        } else {
-            System.out.println("Установите облегченную версию приложения для Android по ссылке");
+        if (clientDeviceYear <= 2015) {
+            if (clientOS == 0)
```
теперь у тебя вложенность есть. Должна быть такая конструкция с 4 вариантами:
```java
if (условие1) {
   // ...
} else if (условие2) {
   // ...
} else if (условие3) {
   // ...
} else {
   // ...
}
```

AndervolV committed:
HW-5 Странно я вроде отправлял этот пулл реквест. А сейчас смотрю на гите нет  его

AndervolV committed:
HW-5 Отправляю
